### PR TITLE
NDRS-899: Set ulimit

### PR DESCRIPTION
**Pull request by marc-casperlabs**
To address issues with networks with > 400 connections, set the open files limit (`ulimit -n`) at the startup.

If this fails, settle for the closest possible maximum. This code will never panic or abort, but warn if the limit cannot be increased.

### CODE REVIEW

**Review by EdHastingsCasperLabs — APPROVED** [review]

File: node/src/reactor.rs [hunk 1]
```diff
@@ -81,6 +82,9 @@ static MEM_DUMP_THRESHOLD_MB: Lazy<Option<u64>> = Lazy::new(|| {
         .ok()
 });
 
+/// The desired limit for open files.
+const TARGET_OPEN_FILES_LIMIT: Limit = 64_000;
```
Comment (goral09) [inline node/src/reactor.rs:86]: We should probably make it configurable via an env variable.

File: node/src/utils/rlimit.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    #[test]
+    fn get_and_reset_ulimit() {
+        // Retrieve limit and set the exact same limit again.
+        let limit = ResourceLimit::<OpenFiles>::get().expect("could not retrieve initial limit");
+        println!("{:?}", limit);
+
+        // Note: We could change it to something different, but we do not want to risk influencing
+        // other tests, so this is safest.
+        limit.set().expect("could not re-set limit");
```
Comment (goral09) [inline node/src/utils/rlimit.rs:143]: We could always change it back to the original limit? Cargo tests run sequentially, right?

**Review by goral09 — COMMENTED** [review]
Really educational PR. I will re-read and try to learn from it :)

**Pull request merged**